src/internal/modules/billing/lib/mappers.js

```diff
@@ -56,40 +47,30 @@ const getTransactionTotals = transactions => {
   return mapValues(totals, val => val.toNumber());
 };
 
-const isMinimimChargeTransaction = trans => trans.isMinimumCharge;
-const isNotMinimumChargeTransaction = trans => !isMinimimChargeTransaction(trans);
+const getSortKey = trans => `${get(trans, 'chargeElement.id')}_${trans.isCompensationCharge ? 1 : 0}`;
 
-const mapTransactionGroup = transactions => ({
-  chargeElement: transactions[0].chargeElement,
-  transactions: transactions.map(mapTransaction),
-  totals: getTransactionTotals(transactions)
+const mapTransaction = trans => ({
+  ...trans,
+  agreements: trans.agreements.map(agreementsMapper.mapAgreement)
 });
 
-const getTransactionGroups = transactions => {
-  const arr = transactions.filter(isNotMinimumChargeTransaction);
-  const grouped = groupBy(arr, trans => trans.chargeElement.id);
-  return Object.values(grouped).map(mapTransactionGroup);
-};
-
 /**
-   *
-   * @param {Object} invoice - payload from water service invoice detail call
-   * @param {Map} documentIds - map of licence numbers / CRM document IDs
-   */
-const mapInvoiceLicences = invoice =>
-  invoice.invoiceLicences.map(invoiceLicence => {
-    const { licenceNumber, id: licenceId } = invoiceLicence.licence;
-    const { id, hasTransactionErrors, transactions } = invoiceLicence;
-
-    return {
-      id,
-      licenceNumber,
-      hasTransactionErrors,
-      link: `/licences/${licenceId}`,
-      minimumChargeTransactions: transactions.filter(isMinimimChargeTransaction),
-      transactionGroups: getTransactionGroups(transactions)
-    };
-  });
+ * Map invoice to view model
+ *
+ * @param {Object} invoice - payload from water service invoice detail call
```

Comment:
Is this JSDoc correct? I thought it wants `invoiceLicence` rather than `invoice`?